demos/powersync-supabase-yjs-text-collab-demo/src/components/providers/SystemProvider.tsx

```diff
@@ -0,0 +1,57 @@
+'use client';
+
+import _ from 'lodash';
+import React, { Suspense } from 'react';
+import Logger from 'js-logger';
+import { PowerSyncContext } from '@journeyapps/powersync-react';
+import { WASQLitePowerSyncDatabaseOpenFactory } from '@journeyapps/powersync-sdk-web';
+import { AppSchema } from '@/library/powersync/AppSchema';
+import { SupabaseConnector } from '@/library/powersync/SupabaseConnector';
+import { useRouter } from 'next/navigation';
+import { CircularProgress } from '@mui/material';
+
+const SupabaseContext = React.createContext<SupabaseConnector | null>(null);
+export const useSupabase = () => React.useContext(SupabaseContext);
+
+const powersync = new WASQLitePowerSyncDatabaseOpenFactory({
+  dbFilename: 'powersync2.db',
+  schema: AppSchema,
+  flags: {
+    disableSSRWarning: true
+  }
+}).getInstance();
+
+export const SystemProvider = ({ children }: { children: React.ReactNode }) => {
+  const [connector] = React.useState(new SupabaseConnector());
+  const [powerSync] = React.useState(powersync);
```

Comment:
This useState probably is not necessary anymore

Reply:
I changed this back to match the other demos now that the underlying issue is fixed.